pkg/syncer/metadatasyncer.go

```diff
@@ -2861,20 +2868,25 @@ func policyQuotaCRAdded(obj interface{}) {
 					cnsoperatorv1alpha1.CnsStoragePolicyUsageSingular, namespace, err)
 				return
 			}
-			foundPvcUsageInstance := false
+			foundPvcUsageInstance = false
+			if isStorageQuotaM2Enabled {
```

Comment:
Is this check necessary? foundSnapUsageInstance should be false if the feature gate is not enabled as well.